libraries/azure_dns_zones_resources.rb

```diff
@@ -0,0 +1,69 @@
+require 'azure_generic_resources'
+
+class AzureDNSZonesResources < AzureGenericResources
+  name 'azure_dns_zones_resources'
+  desc 'Verifies settings for Azure DNS ZONES'
+  example <<-EXAMPLE
+    azure_dns_zones_resources  do
+      it{ should exist }
+    end
+  EXAMPLE
+
+  attr_reader :table
+
+  def initialize(opts = {})
+    # require "byebug"
+    # byebug
+    # Options should be Hash type. Otherwise Ruby will raise an error when we try to access the keys.
+    raise ArgumentError, 'Parameters must be provided in an Hash object.' unless opts.is_a?(Hash)
+
+    # Azure REST API endpoint URL format listing the all resources for a given subscription:
+    #   GET https://management.azure.com/subscriptions/{subscriptionId}/
+    # providers/Microsoft.Network/dnszones?api-version=2018-05-01
+    #
+    # or in a resource group only
+    #   GET https://management.azure.com/subscriptions/{subscriptionId}/
+    #     # providers/Microsoft.Network/dnszones?api-version=2018-05-01
+    #
+    # The dynamic part that has to be created for this resource:
+    #   Microsoft.Network/dnszones?api-version=2019-12-01
+    #
+    # Parameters acquired from environment variables:
+    #   - {subscriptionId} => Required parameter. It will be acquired by the backend from environment variables.
+    #
+    # For parameters applicable to all resources, see project's README.
+    #
+    # User supplied parameters:
+    #   - resource_group => Optional parameter.
+    #   - api_version => Optional parameter. The latest version will be used unless provided.
+    #
+    # Following resource parameters have to be defined/created here.
+    #   resource_provider => Microsoft.Network/dnszones
+
+    opts[:resource_provider] = specific_resource_constraint('Microsoft.Network/dnszones', opts)
+    # static_resource parameter must be true for setting the resource_provider in the backend.
+    super(opts, false)
+
+    # Check if the resource is failed.
+    # It is recommended to check that after every usage of superclass methods or API calls.
+    return if failed_resource?
+
+    # Define the column and field names for FilterTable.
+    #   - column: It is defined as an instance method, callable on the resource, and present `field` values in a list.
+    #   - field: It has to be identical with the `key` names in @table items that will be presented in the FilterTable.
+    # @see https://github.com/inspec/inspec/blob/master/docs/dev/filtertable-usage.md
+    table_schema = [
+      { column: :names, field: :name },
+      { column: :types, field: :type },
+      { column: :ids, field: :id },
+      { column: :locations, field: :location },
```

Comment:
missed properties here

Reply:
[Properties](https://docs.microsoft.com/en-us/rest/api/dns/zones/list#zone) has various other useful attributes .. could we add all of them are give access to properties ideally both would be better

Reply:
this is still pending

Reply:
Done.